### ISSUE TITLE
Editor Team - First Release Coordination 🙌🎉

### ISSUE THREAD

**Issue by emilyashley** [publiclab/PublicLab.Editor#532]
Happy Monday! Looks like everyone is hard at work --  I’m interested to see what pulling together some of our changes looks like for pushing out our first release together as team :tada: 

Curious what the release process looks like? Me, too — I’m also new to this project. Here's a mostly blank issue for us to sort out our thoughts and gather existing documentation. We can update this description as we go!

**Let’s start by:** 

1. Identify one thing you’ve done or solved that you’d like to see in our first release and  

2. the steps we’d take to get it represented in a new release of the editor (or corresponding) repository.

And then we can work thru the second part together and coordinate a release with work from each team member! 

(And maybe after even gather notes on how to improve the process for our second release? Think of this first one as a simplified dress rehearsal with one item from each member to iron out the process, our next release will probably be slightly bigger and more complex 🙌) 

------
## Issues to Track for Release 1 📋

- [ ] Popup Disappearing problem (jywarren/woofmark#3)
- [ ] Drag and drop CSS (jywarren/woofmark#2)
- [ ] Choose File button UI improvement #462
- [ ] Remove file button added #472
- [ ] Fixed toolbar #522
- [ ] Bug report button and template #530
- [ ] Feature to change heading back into plain text (jywarren/woofmark#5)
- [ ] UI testing with jest-puppeteer #541 

## Blockers to Releasing 

- [ ]  #

**Comment by emilyashley** [comment]
cc: @shreyaa-sharmaa @Shulammite-Aso @NitinBhasneria @keshav234156 @jywarren ✨

**Comment by jywarren** [comment]
This is great, thanks @emilyashley!! 🎉 

One thing that may help is to actually map out a flow chart of dependencies for this set of repos, maybe using a shared Google Presentation? I suggested a flow in these comments we can reference to build on:

https://github.com/publiclab/PublicLab.Editor/issues/490#issuecomment-637765410

https://github.com/publiclab/PublicLab.Editor/issues/490#issuecomment-642873893

So mapping how each is tied to an upstream or downstream repo through NPM publication (using http://semver.org/ version numbering) will help us map out the required sequence to get new code merged and included down all the way to `plots2` and then live on https://publiclab.org. 

One thing to bear in mind is how `semver` conventions for versioning/releases work, which i can very roughly summarize as follows:

https://semver.org has guidance for versioning which we could generally follow; roughly:

`0.0.x` are bugfixes
`0.x.0` are new features
`x.0.0` are breaking changes that affect downstream use

so, this lets us make "promises" so to speak (not JavaScript Promises, just the regular kind 😄 ) to downstream libraries that depend on our code, that we won't change any current features they may be relying on without considering the effects and issuing a "breaking" `x.0.0` change. And also that if they see an increase of `0.0.1` they'll know it's just a bugfix and safe to merge. They actually indicate this in `package.json` by specifying a version of the package that accepts bugfix level updates, with a `~` character, so `~0.0.1` (for example). There's a lot more on this in `npm` docs here: https://docs.npmjs.com/files/package.json#dependencies

Sorry for all the talking! And some of you may be familiar with all this already, but just so we're all on the same page! Thanks a ton!!!

**Comment by emilyashley** [comment]
Once we have a list of the issues we want to include in the first release, we can keep track of them using 

- [ ] checkmarks

in this issue description so we know what needs to be done!

**Comment by keshav234156** [comment]
ok, So far I have made following PR:
1. Bold bug fix(https://github.com/jywarren/woofmark/pull/4)
2. Linting tool (https://github.com/publiclab/PublicLab.Editor/pull/519)
3. Custom Insert module ( https://github.com/publiclab/PublicLab.Editor/pull/538)

I would like to see all of them in next first major release.

**Comment by NitinBhasneria** [comment]
I have made this following PR's:
Pending for reviews:
- Popup Disappearing problem solved (https://github.com/jywarren/woofmark/pull/3)
- Drag and drop CSS (https://github.com/jywarren/woofmark/pull/2)
- Add "Related work" section to Title module #534
- Main Image box CSS improved #467 
Merged Work:
- Choose File button UI improvement #462
- Remove file button added #472
I think this all can be included in first release. Thank you all.

**Comment by jywarren** [comment]
Hi all, thanks a lot! Would love to hear from @Shulammite-Aso and @shreyaa-sharmaa here as well!

Just noting -- in https://github.com/publiclab/PublicLab.Editor/pull/530#issuecomment-648302779, i made a comment about how an HTML change to the example templates in this repo would need to be copied over into the corresponding HTML code in the `plots2` repo for the feature to appear there. And would need to be coordinated with the bumping of the next version of the Editor so that the CSS files are pulled in from the new version of the Editor, at the moment the `plots2` HTML changes go live. This takes some good and careful textual warnings and "checklists" in the PRs so that when they're merged we know what has to happen first or immediately after, but it's part of version management so I thought you'd appreciate seeing how it works!

Thanks, all!

**Comment by jywarren** [comment]
Hi folks working on `woofmark` -- so, once the `woofmark` changes are ready and merged into the `jywarren/woofmark` `plots2` branch, we should point this Editor repo's `package.json` at the latest commit in that forked version of `woofmark` in a new PR. That will pull all our `woofmark` changes into this repo, in preparation for a new release here. Does that make sense? Thanks! I'll be reviewing all the `woofmark` changes today too!

**Comment by jywarren** [comment]
Also just checking our recent commit history -- https://github.com/publiclab/PublicLab.Editor/commits/ shows contributions from each of the four team members! 

And on April 14th, https://github.com/publiclab/PublicLab.Editor/commit/65915b4da7d5ca4cbc91077015e5644760434ce4, we released `v2.0.6`. Would this next release, given that it: (a) doesn't break anything downstream and (b) adds some (big and small) new changes, be `v2.1.0`?

**Comment by Shulammite-Aso** [comment]
Hi all!!

Will be happy to see:
Fixed toolbar  #522  and
Bug report button and tamplate #530 in our first release

**Comment by shreyaa-sharmaa** [comment]
Hey everyone! 
My work so far:
1. #521 Decenter feature 
2. https://github.com/jywarren/woofmark/pull/5 Feature to change heading back into plain text
3. #541 UI testing with jest-puppeteer
Would love to see it in the first release!

**Comment by jywarren** [comment]
OK! I believe everyone has at least one if not several PRs merged into the main branch! Let's:

1. open a new PR which increases the `package.json` version number to `v2.1.0`
2. compile up the already merged PR numbers/links and titles in the above PR
3. merge it and ping me, and I'll run `npm publish` to publish a new release!
4. then we'll watch for @dependabot to open a new PR on `plots2` to pull in our new work 🎉 

Someone want to kick this process off? Thanks, all!

**Comment by ebarry** [comment]
Hi everyone 🤩☀️!
Did this process get kicked off? If so, let me know here, if not, maybe today's the day? 🐣

**Comment by jywarren** [comment]
Yes please. I'll choose someone randomly -- I'd like to ask @shreyaa-sharmaa to take point on this, please! And as we do more releases, we can rotate the responsibility! Don't worry, this is just the first one. Thanks!!! 🎉

**Comment by shreyaa-sharmaa** [comment]
Here it is everyone!!!! #555 :heart: :rocket: 
Thank you! :smile:

**Comment by jywarren** [comment]
OK we are getting there!!! https://github.com/publiclab/PublicLab.Editor/pull/555

Actually, @shreyaa-sharmaa above is also a good place to pull in list items for the release notes!

**Comment by jywarren** [comment]
We need to tidy up and update this issue! But, noting `v2.1.1` here: Bump Version from 2.1.1 to 2.1.2, switch to woofmark plots2 branch publiclab/PublicLab.Editor#557 / https://github.com/publiclab/plots2/pull/8285

**Comment by cesswairimu** [comment]
This was successfully merged at https://github.com/publiclab/plots2/pull/8286. Closing this. Thanks everyone
